src/shared_modules/content_manager/tests/unit/CtiDownloader_test.cpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Wazuh Content Manager - Unit Tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * Dec 01, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "CtiDownloader_test.hpp"
+#include "CtiDownloader.hpp"
+#include "HTTPRequest.hpp"
+#include "updaterContext.hpp"
+#include "gtest/gtest.h"
+#include <memory>
+
+const auto OK_STATUS = R"([{"stage":"CtiDummyDownloader","status":"ok"}])"_json;
+const auto FAIL_STATUS = R"([{"stage":"CtiDummyDownloader","status":"fail"}])"_json;
+
+constexpr auto CONTENT_TYPE {"raw"};
+constexpr auto FAKE_CTI_URL {"http://localhost:4444/snapshot/consumers"};
+constexpr auto RAW_URL {"http://localhost:4444/raw"};
+
+/**
+ * @class CtiDummyDownloader
+ *
+ * @brief Dummy class for testing purposes.
+ *
+ */
+class CtiDummyDownloader final : public CtiDownloader
+{
+private:
+    /**
+     * @brief Stores the CTI base parameters.
+     *
+     * @param context Updater context.
+     */
+    void download(UpdaterContext& context) override
+    {
+        const auto parameters {
+            getCtiBaseParameters(context.spUpdaterBaseContext->configData.at("url").get_ref<const std::string&>())};
+        m_lastOffset = parameters.lastOffset;
+        m_lastSnapshotLink = parameters.lastSnapshotLink;
+    }
+
+    int m_lastOffset;               ///< Last offset downloaded from CTI.
```

Comment:
improve: missing initialization

```suggestion
    int m_lastOffset{};               ///< Last offset downloaded from CTI.
```

Reply:
Done!